runtime/bcutil/cfreader.c

```diff
@@ -3601,3 +3606,19 @@ sortMethodIndex(J9CfrConstantPoolInfo* constantPool, J9CfrMethod *list, IDATA st
 		}
 	}
 }
+
+#if JAVA_SPEC_VERSION >= 15
+I_32
+checkClassBytes(J9VMThread *currentThread, U_8* classBytes, UDATA classBytesLength, U_8* segment, U_32 segmentLength) 
+{
+	I_32 rc = 0;
+	U_32 cfrFlags = BCT_JavaMaxMajorVersionShifted | BCT_AnyPreviewVersion | BCT_BasicCheckOnly;
+	PORT_ACCESS_FROM_VMC(currentThread);
+	if (NULL == classBytes) {
+		return rc;
+	}
+	rc = j9bcutil_readClassFileBytes(PORTLIB, NULL, classBytes, classBytesLength, segment, segmentLength, cfrFlags, NULL, NULL, 0, 0);
```

Comment:
Can this not find the class name and return it as a string?  It saves trying to parse it out on the java side

Reply:
It can get the class name, but I don't think we can do the following checks inside j9bcutil_readClassFileBytes before getting the class name. 

```
if (!(thisClass instanceof Type)) {
        throw new ClassFormatError();
}
Type type = (Type)thisClass;
if (!type.getDescriptor().startsWith("L")) {
	throw new ClassFormatError();
}
```

Reply:
If it returns the name though, you can still do the checks in Java without having to partially parse the classfile again